### ISSUE TITLE
Length of contrast vectors depends on high cutoff frequency

### ISSUE THREAD

**Issue by alpinho** [nistats/nistats#354]
The length of the contrast vectors in [this example](https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py), lines 156 and 157,

https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L156
https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L157

depends on the high cutoff frequency set by the user:
https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L99

Length of the contrast vectors should only depend on the number of regressors-of-interest. Regressors-of-no-interest should be introduced automatically, according to the parameters provided.

This issue is in reference to the PR #351

**Comment by bthirion** [comment]
Thx for raising it.
Do you propose an automated zero-padding of contrasts ?

**Comment by alpinho** [comment]
Yes, an automated zero-padding when the high cutoff frequency is increased and the opposite when it is decreased. I suggest an automated adjustment of the length by adding and removing zero elements in the end of the array.

**Comment by bthirion** [comment]
I think that we should not manipulate silently user-provided information. At least issue an explicit warning  when doing it.
Actually, a more elegant solution ---currently available in nistats ! --- is to specify the contrasts symbolically as 'c1 - c2'.
The first thing to do is thus to see if we can improve the examples using this approach.

**Comment by alpinho** [comment]
Ok. Shall this be addressed in PR #351 or separately?

**Comment by bthirion** [comment]
Separately. Best, B
